app/models/miq_worker.rb

```diff
@@ -330,6 +331,17 @@ def self.close_pg_sockets_inherited_from_parent
     end
   end
 
+  # Close all open DRb connections so that connections in the parent's memory space
+  # which is shared due to forking the child process do not pollute the child's DRb
+  # connection pool.  This can lead to errors when the children connect to a server
+  # and get an incorrect response back.
+  #
+  # ref: https://bugs.ruby-lang.org/issues/2718
+  def self.close_drb_pool_connections
+    require 'drb/drb'
```

Comment:
`#MAJOR HACK approaching...`

It's a stable api since it's been there [since at least 2003](https://github.com/ruby/ruby/commit/151f1241c665b0307234e931bec2c32bfea0138d#diff-e979bf2f831d9826629559b8628809e9)  😆 🤣